src/backend/src/lib.rs

```diff
@@ -394,6 +402,36 @@ fn list_user_tokens() -> Vec<Token> {
     read_state(|s| s.user_token.get(&stored_principal).unwrap_or_default().0)
 }
 
+/// Adds a new custom token to the user.
+#[update(guard = "caller_is_not_anonymous")]
+fn add_user_custom_token(token: UserToken) {
+    let stored_principal = StoredPrincipal(ic_cdk::caller());
+
+    let find = |t: &UserToken| -> bool { t.clone() == token };
+
+    mutate_state(|s| add_to_user_token(stored_principal, &mut s.user_custom_token, &token, &find));
+}
+
+#[update(guard = "caller_is_not_anonymous")]
+fn remove_user_custom_token(token_id: UserTokenId) {
+    let stored_principal = StoredPrincipal(ic_cdk::caller());
+
+    let find = |t: &UserToken| -> bool { UserTokenId::from(t.clone()) == token_id };
+
+    mutate_state(|s| remove_from_user_token(stored_principal, &mut s.user_custom_token, &find));
+}
+
```

Comment:
I personally would make the query below a certified query but I see that it is the same as the query above.  And who knows, there are shiny new certification methods coming that don't require you to keep all your data in a merkle tree!